src/lifecycle/run.rs

```diff
@@ -37,13 +37,28 @@ use {
 /// On desktop platforms, this yields control to a simple game loop controlled by a Timer. On wasm,
 /// this yields control to the browser functions setInterval and requestAnimationFrame
 pub fn run<T: State>(title: &str, size: Vector, settings: Settings) {
-    if let Err(error) = run_impl::<T>(title, size.into(), settings) {
+    if let Err(error) = run_impl::<T, _>(title, size.into(), settings, || T::new()) {
+        T::handle_error(error);
+    }
```

Comment:
Couldn't this be written as `run_with(title, size, settings, || T::new())`? Less copy-pasting then

Reply:
Good point. I guess the extra layer on the stack doesn't really matter for a function that is only ever run once.